iocage/lib/Datasets.py

```diff
@@ -0,0 +1,167 @@
+            self.root = self._get_or_create_dataset(
+                "iocage",
+                root_name=pool.name,
+                pool=pool
+            )
+            return
+
+        active_pool = self.active_pool
+
+        if active_pool is None:
+            msg = ("iocage is not activated yet - "
+                   "please run `iocage activate` first and select a pool")
+            self.logger.error(msg)
+            raise Exception(msg)
+        else:
+            self.root = self.zfs.get_dataset(f"{active_pool.name}/iocage")
+
+    @property
+    def active_pool(self):
+        for pool in self.zfs.pools:
+            if self._is_pool_active(pool):
+                return pool
+        return None
+
+    @property
+    def releases(self):
+        return self._get_or_create_dataset("releases")
+
+    @property
+    def base(self):
+        return self._get_or_create_dataset("base")
+
+    @property
+    def jails(self):
+        return self._get_or_create_dataset("jails")
+
+    @property
+    def logs(self):
+        return self._get_or_create_dataset("log")
+
+    def activate(self, mountpoint=None):
+        self.activate_pool(self.root.pool, mountpoint)
+
+    def activate_pool(self, pool, mountpoint=None):
+
+        if self._is_pool_active(pool):
+            msg = f"ZFS pool '{pool.name}' is already active"
+            self.logger.warn(msg)
+
+        if not isinstance(pool, libzfs.ZFSPool):
+            msg = "Cannot activate invalid ZFS pool"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+        if pool.status == "UNAVAIL":
+            msg = f"ZFS pool '{pool.name}' is UNAVAIL"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+        other_pools = filter(lambda x: x.name != pool.name, self.zfs.pools)
+        for other_pool in other_pools:
+            self._deactivate_pool(other_pool)
+
+        self._activate_pool(pool)
+
+        root_dataset_args = {
+            "pool": pool
+        }
+
+        if mountpoint is not None:
+            root_dataset_args["mountpoint"] = mountpoint
+
+        self.root = self._get_or_create_dataset(
+            "iocage",
+            **root_dataset_args
+        )
+
+    def _is_pool_active(self, pool):
+        prop = self.ZFS_POOL_ACTIVE_PROPERTY
+        return self._get_pool_property(pool, prop) == "yes"
+
+    def _get_pool_property(self, pool, prop):
+        try:
+            return pool.root_dataset.properties[prop].value
+        except:
+            return None
+
+    def _get_dataset_property(self, dataset, prop):
+        try:
+            return dataset.properties[prop].value
+        except:
+            return None
+
+    def _activate_pool(self, pool):
+        self._set_pool_activation(pool, True)
+
+    def _deactivate_pool(self, pool):
+        self._set_pool_activation(pool, False)
+
+    def _set_pool_activation(self, pool, state):
+        prop = self.ZFS_POOL_ACTIVE_PROPERTY
```

Comment:
I'm didn't make the decision to limit the code width to 80 characters, but I respect it. I would like to change it, because I don't want to write different code because of such a virtual limitation. Can be addressed in a future refactoring iteration.